src/ui/public/dialogues/safe_confirm.factory.js

```diff
@@ -0,0 +1,11 @@
+import uiModules from 'ui/modules';
+import { safeConfirm } from './safe_confirm';
+import './directives/confirm_dialogue';
+import './modal_dialogue.factory';
+
+const module = uiModules.get('kibana');
+
+module.factory('safeConfirm', function ($timeout, ModalDialogue, Promise) {
+  return (message, confirmButtonText = 'Okay', cancelButtonText = 'Cancel') =>
+    $timeout(() => safeConfirm(message, confirmButtonText, cancelButtonText, Promise, ModalDialogue));
```

Comment:
I should have kept the original comment.  I'm not 100% sure myself, but sounded necessary.  Here is the original comment:

```
/*
 * Angular doesn't play well with thread blocking calls such as
 * unless those calls are specifically handled inside a call
 * to $timeout(). Rather than litter the code with that implementation
 * detail, safeConfirm() can be used.
 *
 * WARNING: safeConfirm only blocks the thread beginning on the next tick. For that reason, a
 * promise is returned so consumers can handle the control flow.
 *
 * Usage:
 *  safeConfirm('This message will be shown in a modal dialog').then(
 *    function () {
 *      // user clicked the okay button
 *    },
 *    function () {
 *      // user canceled the confirmation
 *    }
 *  );
 */
```

It's possible it's not even necessary since we are using promises and not a thread blocking call.